test/stripe/converter_test.exs

```diff
@@ -0,0 +1,128 @@
+defmodule Stripe.ConverterTest do
+  use ExUnit.Case
+
+  alias Stripe.Converter
+
+  defmodule Person do
+    defstruct [:email, :first_name, :last_name, :legal_entity, :metadata]
+    def relationships, do: %{}
+  end
+
+  defmodule PersonWithMissingRelationship do
+    defstruct [:card]
+    def relationships, do: %{card: Stripe.ConverterTest.CreditCard}
+  end
+
+  defmodule AuthToken do
+    defstruct [:id, :card, :client_ip, :created, :livemode, :type, :used]
+    def relationships, do: %{
+      card: Stripe.ConverterTest.CreditCard,
+      created: DateTime
+    }
+  end
+
+  defmodule CreditCard do
+    defstruct [:brand, :country, :exp_month]
+    def relationships, do: %{}
+  end
+
+  test "converts a Stripe response into a struct" do
+    expected_result = %Stripe.ConverterTest.Person{
+      first_name: "Leslie",
+      last_name: "Knope",
+      email: "knope@stripe.com",
+      metadata: %{},
+      legal_entity: %{
+        address: %{
+          city: "Pawnee",
+          country: "US",
+          state: "IN"
+        },
+        business_name: "Parks and Rec",
+        dob: %{
+          day: 23,
+          month: 12,
+          year: 2016
+        }
+      }
+    }
+
+    result = Converter.stripe_map_to_struct(Stripe.ConverterTest.Person, json_response)
```

Comment:
Thoughts on aliasing `Stripe.ConverterTest.Person`? It may make this a bit easier to read through.